### ISSUE TITLE
Flake and GHC 9.2.2

### ISSUE THREAD

**Issue by jmatsushita** [turion/rhine#162]
Hi there,

I've been using nix flakes to setup my local development environment with cabal only and have used that setup to get rhine to work with GHC 9.2.2. Is that something you're interested in a PR for? I've been using this setup to work on the `terminal` and `vty` examples I mentioned in https://github.com/turion/rhine/issues/161

I'm also happy to try and break this down into 2 separate PRs (one to upgrade GHC, and another to port the project to flakes). It'll take a bit of work to make sure the CI also works properly as well as your local dev environment (I'm on an M1 mac).

Cheers,

Jun

**Comment by turion** [comment]
Yes that sounds like a great idea! I just recently started using flakes, that's why I hadn't added anything like it yet.

> I'm also happy to try and break this down into 2 separate PRs

Yes, that's the best way to go. Thanks!

**Comment by jmatsushita** [comment]
I gave a shot at a simple flake. Since I've worked on the 9.2.2 upgrade using flakes, I guess it makes sense to let the dust settle on that first :)
